arch/cpu/native/net/tun6-net.c

```diff
@@ -95,35 +104,144 @@ ssystem(const char *fmt, ...)
+static bool
+set_default_prefix(const char *prefix)
+{
+  char *ipaddr = strdup(prefix);
+  if(!ipaddr) {
+    return false;
+  }
+
+  uip_ip6addr_t prefix_addr;
+  bool success = false;
+  char *s = strchr(ipaddr, '/');
+  if(s) {
+    *s = '\0';
+    if(uiplib_ipaddrconv(ipaddr, &prefix_addr)) {
+      uip_ds6_set_default_prefix(&prefix_addr);
+      success = true;
+    }
+  }
+  free(ipaddr);
+  return success;
+}
+/*---------------------------------------------------------------------------*/
+const char *
+tun6_net_get_prefix(void)
+{
+  return config_ipaddr;
+}
+/*---------------------------------------------------------------------------*/
+void
+tun6_net_set_prefix(const char *prefix)
+{
+  if(!set_default_prefix(prefix)) {
+    LOG_WARN("Failed to set default prefix %s\n", prefix);
+  }
+  config_ipaddr = prefix;
+}
+/*---------------------------------------------------------------------------*/
+const char *
+tun6_net_get_tun_name(void)
+{
+  return config_tundev;
+}
+/*---------------------------------------------------------------------------*/
+void
+tun6_net_set_tun_name(const char *tun_name)
+{
+  /* Ignore "/dev/" if present in tun device name */
+  if(strncmp("/dev/", tun_name, 5) == 0) {
+    tun_name += 5;
+  }
+  strncpy(config_tundev, tun_name, sizeof(config_tundev) - 1);
+  config_tundev[sizeof(config_tundev) - 1] = '\0';
+}
+/*---------------------------------------------------------------------------*/
+int
+tun6_net_get_mtu(void)
+{
+  return config_mtu;
+}
 /*---------------------------------------------------------------------------*/
 void
-ifconf_cleanup(const char *dev)
+tun6_net_set_mtu(int mtu_size)
+{
+  config_mtu = MAX(MIN_MTU_SIZE, mtu_size);
+}
+/*---------------------------------------------------------------------------*/
+static int
+tun_dev_callback(const char *optarg)
+{
+  tun6_net_set_tun_name(optarg);
+  return 0;
+}
+CONTIKI_OPTION(TUN_PRIO, { "t", required_argument, NULL, 0 },
+               tun_dev_callback,
+               "name of tun interface (default: " DEFAULT_TUN ")\n");
+
+/*---------------------------------------------------------------------------*/
+static int
+prefix_callback(const char *optarg)
+{
+  tun6_net_set_prefix(optarg);
+  return 0;
+}
+CONTIKI_OPTION(TUN_PRIO + 1, { "prefix", required_argument, NULL, 0 },
+               prefix_callback,
+               "Subnet prefix (default: " DEFAULT_PREFIX ")\n");
+/*---------------------------------------------------------------------------*/
+static int
+mtu_callback(const char *optarg)
+{
+  tun6_net_set_mtu(atoi(optarg));
+  return 0;
+}
+CONTIKI_OPTION(TUN_PRIO + 2, { "mtu", required_argument, NULL, 0 },
+               mtu_callback, "interface MTU size\n");
+/*---------------------------------------------------------------------------*/
+static void
+cleanup(void)
 {
 #define TMPBUFSIZE 128
   /* Called from signal handler, avoid unsafe functions. */
   char buf[TMPBUFSIZE];
+#ifndef __APPLE__
```

Comment:
This duplicates the entire strncat-block, but the start of the block is identical, and the "ifconfig device down" command is also identical. It would be nice to not duplicate that code.

Reply:
I kept the OS blocks separated because I thought it looked messy to mix them, but I agree that it is nice to avoid code duplication. I updated this.